### PR TITLE
Compatibility: Add workaround for Ezoic

### LINKED ISSUE #13340: Ezoic Incompatibility with Web Stories
## Bug Description

The Web Stories checklist displays a compatibility warning whenever Ezoic plugin is just activated & WordPress Integrated  

This is a diffchecker link to compare the removals/additions after activating Ezoic: https://www.diffchecker.com/B2rrJGd0/
(Left column is Web Stories only, Right column is when Ezoic is activated)

Example URL affected:
https://lucksany.com/web-stories/test-7/

AMP test: https://search.google.com/test/amp/result?id=0DSwtt2fTrpe8IGB2_xAIA

## Expected Behaviour

There are no validation error when Web Stories and Ezoic are activated together.

## Steps to Reproduce

1. Activate Ezoic plugin
2. Open existing Web Story
3. Notice compatibility warning
4. Check AMP validation test

## Additional Context

Similar reports with plugin incompatibility:
https://wordpress.org/support/topic/compatibility-and-404-error
https://wordpress.org/support/topic/tag-amp-story-is-not-allowed-to-have-any-sibling-tags-8/
https://wordpress.org/support/topic/serious-debug-issue

- Plugin Version: 1.32.0
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by AnuragVasanwala**
## Context

The goal of the PR is to add missing html tag `doctype` required for valid AMP and `amp-story-auto-analytics` script tag.

### Ezoic Plugin (Version 2.8.29)

When installing and activating Ezoic plugin, it is observed that `doctype` is not prefixed into outputted story resulting into invalid amp story.

In Ezoic plugin, this has been mentioned and possible solution is provided but commented out:
```php
// File: wp-content/plugins/ezoic-integration/includes/class-ezoic-integrator.php (Line 74-78)

// fixes for AMP validations
/*if ( Ezoic_Integration_Request_Utils::is_amp_endpoint() ) {
    $amp_request = new Ezoic_Amp_Validation($orig_content);
    $orig_content = $amp_request->fix_amp_validation();
}*/
```

**Fix:** This has been fixed by adding `doctype` when missing.

### Missing `amp-story-auto-analytics` script

When default AMP analytics is used, `<amp-story-auto-analytics .../>` is added to the outputted story but script required for `amp-story-auto-analytics` is not appended.

**Fix:** With this PR, when `amp-story-auto-analytics` is added, it will also add required script to the outputted story.

## Relevant Technical Choices

Not applicable.

## User-facing changes

Not applicable.

### Scenario: When default AMP Analytics is used

1. Add Google Analytics ID (for testing you can add `UA-123456789-0`) into `Settings->Google Analytics`
2. Create/edit a story and update anything on the story
3. Save/Update story
4. Open story page and check for AMP Validation error ([install AMP Validator for Google Chrome](https://chrome.google.com/webstore/detail/amp-validator/nmoffdblmcmgeicmolmhobpoocbbmknc))

**Cause:**
This is happening because, when using default AMP Analytics (without Google SiteKit) WebStories adds `amp-story-auto-analytics` tag but it does not add required script for the tag.

### Scenario: Ezoic Plugin

1. Install and Activate Ezoic Plugin
2. Create/edit a story and update anything on the story
3. Save/Update story
4. Open story page and check for AMP Validation error ([install AMP Validator for Google Chrome](https://chrome.google.com/webstore/detail/amp-validator/nmoffdblmcmgeicmolmhobpoocbbmknc))

**Cause:**
This is happening because, `doctype` is not added to the outputted story and resulting AMP story becomes invalid.

- [x] This is a non-user-facing change and requires no QA

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes https://github.com/GoogleForCreators/web-stories-wp/issues/13340

### CODE REVIEW

**Comment by AnuragVasanwala** [comment]
@swissspidy As per the previous sync-up meeting, I have revised implementation by adding a new Ezoic integration which fixes AMP validation error.

I will also report the `OutputBuffer` issue with detailed information to Ezoic support team, as well.

File: includes/Integrations/Ezoic.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	/**
+	 * Determines whether Ezoic is installed and active.
+	 *
+	 * @since 1.33.0
+	 *
+	 * @return bool Whether Ezoic is installed and active.
+	 */
+	protected function is_plugin_installed_active(): bool {
```
Comment (swissspidy) [inline includes/Integrations/Ezoic.php]: Pro tip: implement the `Conditional` interface in this class and rename this method here to `is_needed` and make it static. Then the service will only ever be registered if Ezoic is active. That means no extra check is needed in `register()`

Comment (AnuragVasanwala) [reply includes/Integrations/Ezoic.php]: Done ✅ with commit https://github.com/GoogleForCreators/web-stories-wp/pull/13354/commits/1ffbca798919dc7d2fd8d6a36e7dd65a23344397.

**Review by swissspidy — APPROVED** [review]
Works like a charm!

**Pull request merged**